src/components/ProductDetailsPage/ProductDetailsPage.js

```diff
@@ -1,85 +1,90 @@
-import React, { useState } from 'react';
-
+import React from 'react';
 import ProductImage from '../ProductImage';
 import ProductOrder from '../ProductOrder';
 import Accordion from '../Accordion';
 import ContactDetails from '../Footer/ContactDetails';
 import { accordionItemsData } from '../../config/dataItemsAccordion';
 import { Button } from '@/shared';
 import { ADDED_TO_WISHLIST_NOTIFICATION } from '../../constants/notificationData';
-
 import './ProductDetailsPage.scss';
 
 export const CN = 'product-details';
+class ProductDetailsPage extends React.Component {
+  constructor(props) {
+    super(props);
 
-const ProductDetailsPage = (props) => {
-  const laptopWidth = 1024;
-  const [isVerticalCarousel, setVerticalCarousel] = useState(
-    window.innerWidth > laptopWidth
-  );
-  const {
-    match: {
-      params: { id }
-    },
-    history: { goBack },
-    products
-  } = props;
-
-  const product = products.find(({ _id }) => _id === id);
+    const laptopWidth = 1024;
 
-  window.onresize = () => setVerticalCarousel(window.innerWidth > laptopWidth);
-
-  if (products.length) {
-    const {
-      images, title, price, description
-    } = product;
+    this.state = {
+      isVerticalCarousel: window.innerWidth > laptopWidth
+    };
+  }
 
-    const productImageData = images.map(({ claudinaryId }) => ({
-      src: `${process.env.IMAGE_URL}/${claudinaryId}`
-    }));
+  setVerticalCarousel() {
+    this.state({
+      isVerticalCarousel: window.innerWidth > this.laptopWidth
+    });
+  }
 
-    const addToWishlistWithNotification = () => {
-      const { showMessage, addToWishlist } = props;
-      addToWishlist(id);
-      showMessage(ADDED_TO_WISHLIST_NOTIFICATION);
-    };
+  addToWishlistWithNotification = () => {
```

Comment:
This one probably could be rewritten to method (I mean: addToWishlistWithNotification() {}), please try to rewrite it and check if it works :) In class components it is better to use class methods instead or arrow functions.

Reply:
done)